q2_demux/test/test_demux.py

```diff
@@ -197,95 +183,149 @@ def _compare_manifests(self, act_manifest, exp_manifest):
+                                index=['sample1', 'sample2', 'sample3',
+                                       'sample4', 'sample5'])
+        self.barcode_map = qiime2.MetadataCategory(barcode_map)
+
     def test_valid(self):
-        actual = emp(self.bsi, self.barcode_map)
+        actual = emp_single(self.bsi, self.barcode_map)
         output_fastq = list(actual.sequences.iter_views(FastqGzFormat))
-        # two per-sample files were written
-        self.assertEqual(len(output_fastq), 2)
+        # five per-sample files were written
+        self.assertEqual(len(output_fastq), 5)
 
         # sequences in sample1 are correct
-        sample1_seqs = skbio.io.read(output_fastq[0][1].open(),
-                                     format='fastq',
-                                     phred_offset=33, compression='gzip',
-                                     constructor=skbio.DNA)
-        sample1_seqs = list(sample1_seqs)
-        self.assertEqual(len(sample1_seqs), 2)
-        self._compare_sequence_to_record(sample1_seqs[0], self.sequences[0])
-        self._compare_sequence_to_record(sample1_seqs[1], self.sequences[1])
+        self._validate_sample_fastq(
+            output_fastq[0][1].open(), self.sequences, [0, 5])
 
         # sequences in sample2 are correct
-        sample2_seqs = skbio.io.read(output_fastq[1][1].open(),
-                                     format='fastq',
-                                     phred_offset=33, compression='gzip',
-                                     constructor=skbio.DNA)
-        sample2_seqs = list(sample2_seqs)
-        self.assertEqual(len(sample2_seqs), 2)
-        self._compare_sequence_to_record(sample2_seqs[0], self.sequences[2])
-        self._compare_sequence_to_record(sample2_seqs[1], self.sequences[3])
+        self._validate_sample_fastq(
+            output_fastq[1][1].open(), self.sequences, [2, 4])
+
+        # sequences in sample3 are correct
+        self._validate_sample_fastq(
+            output_fastq[2][1].open(), self.sequences, [1, 3])
+
+        # sequences in sample4 are correct
+        self._validate_sample_fastq(
+            output_fastq[3][1].open(), self.sequences, [7, 10])
+
+        # sequences in sample5 are correct
+        self._validate_sample_fastq(
+            output_fastq[4][1].open(), self.sequences, [6, 8, 9])
 
         # manifest is correct
         act_manifest = list(actual.manifest.view(FastqManifestFormat).open())
         exp_manifest = ['sample-id,filename,direction\n',
                         'sample1,sample1_1_L001_R1_001.fastq.gz,forward\n',
-                        'sample2,sample2_2_L001_R1_001.fastq.gz,forward\n']
+                        'sample3,sample3_2_L001_R1_001.fastq.gz,forward\n',
+                        'sample2,sample2_3_L001_R1_001.fastq.gz,forward\n',
+                        'sample5,sample5_4_L001_R1_001.fastq.gz,forward\n',
+                        'sample4,sample4_5_L001_R1_001.fastq.gz,forward\n']
```

Comment:
@gregcaporaso can you confirm that this manifest looks correct? I shuffled the order of the barcodes, so that we don't see samples in order (in order to invoke the bug with filehandle closure so that I could test it). In particular can you confirm the number between the sample name and the lane?